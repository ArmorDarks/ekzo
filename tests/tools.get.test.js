/* eslint-env jest */

const { renderSync } = require('node-sass')

describe('Tool get', () => {
  it('should get from map', () => {
    const { css } = renderSync({
      data: `
        @import 'tools/get';
        $map: ( prop: value );
        .output { result: ekzo-get($map, prop); }
      `
    })
    expect(css.toString()).toMatchSnapshot()
  })

  it('should get from nested map', () => {
    const { css } = renderSync({
      data: `
        @import 'tools/get';
        $map: (
          depth1: (
            depth2: depth2Value
          )
        );
        .output { result: ekzo-get($map, depth1, depth2); }
      `
    })
    expect(css.toString()).toMatchSnapshot()
  })

  it('should throw on wrong key', () => {
    expect(() => renderSync({
      data: `
        @import 'tools/get';
        $map: (
          depth1: (
            nope: value
          )
        );
        .output { result: ekzo-get($map, depth1, depth2); }
      `
    })).toThrowErrorMatchingSnapshot()
  })

  it('should throw on non-map value', () => {
    expect(() => renderSync({
      data: `
        @import 'tools/get';
        $map: (
          depth1: nope
        );
        .output { result: ekzo-get($map, depth1, depth2); }
      `
    })).toThrowErrorMatchingSnapshot()
  })
})
