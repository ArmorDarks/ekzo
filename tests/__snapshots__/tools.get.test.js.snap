// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tool get should get from map 1`] = `
".output {
  result: value; }
"
`;

exports[`Tool get should get from nested map 1`] = `
".output {
  result: depth2Value; }
"
`;

exports[`Tool get should throw on non-map value 1`] = `"[ekzo-get]: cannot search property \`depth2\` in non-map value, \`string\` provided"`;

exports[`Tool get should throw on wrong key 1`] = `"[ekzo-get]: cannot find property \`depth2\` in specified map"`;
